device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1539,8 +1541,6 @@ private void updateStatus(IotHubConnectionStatus newConnectionStatus, IotHubConn
             {
                 this.multiplexingStateCallback.execute(newConnectionStatus, reason, throwable, this.multiplexingStateCallbackContext);
             }
-
-            this.deviceIOConnectionStatusChangeCallback.execute(newConnectionStatus, reason, throwable, null);
```

Comment:
The user facing state callbacks were above this SDK facing state callback previously. Now this SDK facing state callback executes first